### PR TITLE
Add operation timing logging

### PR DESCRIPTION
**Pull request by reiddraper**
Starting a conversation here. Originally these were cooked up for a quick deploy, so there might be better, cleaner ways of organizing things for a PR. But I wanted to get this started to document that conversation.

### CODE REVIEW

File: src/riak_cs_utils.erl [hunk 1]
```diff
@@ -394,12 +394,21 @@ delete_object(Bucket, Key, RiakcPid) ->
...
         true ->
             term_to_binary(Term, [compressed]);
         false ->
             term_to_binary(Term)
-    end.
+    end,
+    EndTime = os:timestamp(),
+    TimeDiff = timer:now_diff(EndTime, StartTime),
+    Size = byte_size(Binary),
+    lager:debug("term_to_binary took ~B microseconds for ~B bytes, with"
```
Comment (kellymclaughlin) [inline src/riak_cs_utils.erl]:
```
riak_cs_utils.erl:408: Expression produces a value of type 'ok' | {'error','lager_not_running'}, but this value is unmatched
```

Comment (reiddraper) [inline src/riak_cs_utils.erl]: Amended this into the first commit, now cc92bdb8e4810e92d0d069962a55e971ae3897ef.

**Comment by slfritchie** [comment]
Hrm.  What do we know about lager:debug() overhead vs. DTrace overhead vs. Erlang built-in tracing overhead?  If lager's level checking is far more expensive than the others, then I'd recommend the others?

**Comment by reiddraper** [comment]
> Hrm. What do we know about lager:debug() overhead vs. DTrace overhead vs. Erlang built-in tracing overhead? If lager's level checking is far more expensive than the others, then I'd recommend the others?

This is probably worth asking @Vagabond about. I'm under the impression that messages below the current log-level threshold are 'cheap', but for what value of 'cheap', I don't know.

**Comment by Vagabond** [comment]
They are 'cheap' but not free. An ETS (formerly mochiglobal) lookup is done per log message to see if the current log level is elegible for logging.

**Comment by reiddraper** [comment]
@kellymclaughlin any thoughts?

**Comment by reiddraper** [comment]
The current consensus is that this is too risky to ship with until we've done performance testing. We'll have the basho-patches option until then, and the option of looking into how to use Erlang's tracing for a similar effect.

**Comment by Vagabond** [comment]
I'd be interested in hearing more on why this is 'risky'.

**Comment by reiddraper** [comment]
@Vagabond suppose it's still open for debate. @kellymclaughlin and I were thinking that since we didn't fully understand the performance impact, it was a bit rushed to get it in a point release (due to go out in a day or so) without doing our normal performance testing. That being said, I'm open minded about this.

**Comment by kellymclaughlin** [comment]
> @Vagabond suppose it's still open for debate. @kellymclaughlin and I were thinking that since we didn't fully understand the performance impact, it was a bit rushed to get it in a point release (due to go out in a day or so) without doing our normal performance testing. That being said, I'm open minded about this.

Agreed, I doubt there would be any significant performance impact, but without running performance tests it's just conjecture. I'd rather keep it as something to slide into basho-patches until we have time to gather real data and we just don't have that time right now.

**Pull request closed without merging**